### PR TITLE
[VL] Correct hive write table

### PR DESCRIPTION
**Pull request by ulysses-you**
## What changes were proposed in this pull request?

This pr add some additional vanilla Spark config check if we can safely convert hive write to velox write.
- HiveUtils.CONVERT_METASTORE_PARQUET
- HiveUtils.CONVERT_INSERTING_PARTITIONED_TABLE

## How was this patch tested?

Add tests

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @JkSelf

File: backends-velox/src/main/scala/org/apache/spark/sql/hive/execution/HiveFileFormat.scala [hunk 1]
```diff
@@ -77,6 +74,19 @@ class HiveFileFormat(fileSinkConf: FileSinkDesc)
...
+  /**
+   * Gluten only supports convert [[InsertIntoHiveDirCommand]], see [[GlutenColumnarRules]].
+   */
+  private def isGlutenHiveWrite(sparkSession: SparkSession): Boolean = {
+    fileSinkConf.tableInfo
+      .getOutputFileFormatClassName()
+      .equals("org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat") &&
+      sparkSession.sparkContext.conf.getOption("spark.plugins")
+        .contains("io.glutenproject.GlutenPlugin") &&
+      sparkSession.sessionState.conf.getConf(HiveUtils.CONVERT_METASTORE_PARQUET) &&
```
Comment (JkSelf) [inline backends-velox/src/main/scala/org/apache/spark/sql/hive/execution/HiveFileFormat.scala:86]: Do we need to add these configurations check [here](https://github.com/oap-project/gluten/blob/116d3dcdbec819d3fb06514bdd0a3ed03986652a/gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenColumnarRules.scala#L107)?

Comment (ulysses-you) [reply backends-velox/src/main/scala/org/apache/spark/sql/hive/execution/HiveFileFormat.scala:86]: The relation conversion is at post resolution hoc, which is always been applied before columnar rules. So we won't hit the right pattern if the config is unexpected.

**Review by JkSelf — APPROVED** [review]
LGTM

**Pull request merged**